input.js

```diff
@@ -40,20 +43,27 @@ const Input = (() => {
 
     function setMessage(messageHandler) {
         rl.on('line', (message) => {
-		messageHandler(message)
+            messageHandler(message)
+        })
+    }
+
+    function setLeave(leaveHandler) {
+        rl.on('SIGINT', () => {
+            leaveHandler()
         })
     }
 
     function clearLine() {
-	readline.clearLine(process.stdout, 0)
-	readline.cursorTo(process.stdout, 0)
+        readline.clearLine(process.stdout, 0)
+        readline.cursorTo(process.stdout, 0)
```

Comment:
so the clearLine(stream, direction) clears the stream(stdout in out case) in the direction(-1 for left of cursor, 1 for right of cursor, 0 for entire line)